src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2302,34 +2307,91 @@ protected override void HandleSetBreakpointsRequestAsync(IRequestResponder<SetBr
 
         protected override void HandleSetExceptionBreakpointsRequestAsync(IRequestResponder<SetExceptionBreakpointsArguments> responder)
         {
-            List<string> filter = responder.Arguments.Filters;
-            if (m_engineConfiguration.ExceptionSettings.Categories.Count > 0)
+            List<ExceptionFilterOptions> filterOptions = responder.Arguments.FilterOptions;
+            if (filterOptions != null && filterOptions.Count > 0)
             {
-                if (filter == null || filter.Count == 0)
+                foreach (ExceptionFilterOptions filterOption in filterOptions)
                 {
-                    SetAllExceptions(enum_EXCEPTION_STATE.EXCEPTION_STOP_SECOND_CHANCE);
-                }
-                else if (filter.Contains(ExceptionBreakpointFilter.Filter_All))
-                {
-                    enum_EXCEPTION_STATE state = enum_EXCEPTION_STATE.EXCEPTION_STOP_SECOND_CHANCE | enum_EXCEPTION_STATE.EXCEPTION_STOP_FIRST_CHANCE | enum_EXCEPTION_STATE.EXCEPTION_STOP_USER_FIRST_CHANCE;
+                    if (filterOption.FilterId == ExceptionBreakpointFilter.Filter_All)
+                    {
+                        if (!string.IsNullOrWhiteSpace(filterOption.Condition))
+                        {
+                            string[] conditions = filterOption.Condition.Split(',');
+                            ExceptionSettings.CategoryConfiguration category = m_engineConfiguration.ExceptionSettings.Categories.First(); // How to map filterOption.FilterId to Category Name / Guid.
+
+                            enum_EXCEPTION_STATE state = enum_EXCEPTION_STATE.EXCEPTION_STOP_SECOND_CHANCE | enum_EXCEPTION_STATE.EXCEPTION_STOP_FIRST_CHANCE | enum_EXCEPTION_STATE.EXCEPTION_STOP_USER_FIRST_CHANCE;
 
-                    if (filter.Contains(ExceptionBreakpointFilter.Filter_UserUnhandled))
+                            IEnumerable<string> toAdd = conditions.Except(m_exceptionBreakpoints);
+                            foreach (string condition in toAdd)
+                            {
+                                var exceptionInfo = new EXCEPTION_INFO[1];
+                                exceptionInfo[0].dwState = state;
+                                exceptionInfo[0].guidType = category.Id;
+                                exceptionInfo[0].bstrExceptionName = condition;
+                                m_engine.SetException(exceptionInfo);
+
+                                m_exceptionBreakpoints.Add(condition);
+                            }
+
+                            IEnumerable<string> toRemove = m_exceptionBreakpoints.Except(conditions).ToList();
+                            foreach (string condition in toRemove)
+                            {
+                                if (m_exceptionBreakpoints.Contains(condition))
+                                {
```

Comment:
How could this fail?

Reply:
Ah, residual from when I was originally looping through all of the conditions in m_exceptionBreakpoints. 

Will remove